### DISCUSSION TITLE
How to nucleate at grain boundaries instead of random locations

### DISCUSSION THREAD

**Discussion by Sabalei** [idaholab/moose#29600] [Q&A Modules: Phase field] [answered]
Hi moose community,
I am trying to simulate static recrystallization.
I am trying to follow the https://mooseframework.inl.gov/modules/phase_field/Nucleation/DiscreteNucleation.html instructions below:

"Direct order parameter modification
...

I want to know how to control nucleation at grain boundaries rather than random locations

**Comment by GiudGiud** [comment, marked as answer]
I think you'll want to modify DiscreteNucleation. You can create a new object called GrainBoundaryDiscreteNucleation. 
The first step there will be to detect grain boundaries, which you can do by analyzing the order parameters
